source/common/upstream/upstream_impl.cc

```diff
@@ -983,6 +983,12 @@ bool BaseDynamicClusterImpl::updateDynamicHostList(
         hosts_changed = true;
       }
 
+      // Did the priority change?
+      if (host->priority() != existing_host->second->priority()) {
+        existing_host->second->priority(host->priority());
+        hosts_changed = true;
```

Comment:
We already have a test case for a host moving between priorities: https://github.com/envoyproxy/envoy/pull/4507/files#diff-6dc8e188029fa504fc2e036afcbdb1fdR572

I added asserts to it to verify that we're correctly updating the the priority of the host in this case. I'll update the description

Reply:
But doesn't this code make it a functional change also because you are setting `hosts_changed` to true? How did it work before or it was covered by some other branch?